caikit/core/model_management/multi_model_finder.py

```diff
@@ -0,0 +1,137 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The MultiModelFinder configures a set of other model finders that will be used
+in sequence to try loading models.
+
+Configuration for MultiModelFinder lives under the config as follows:
+
+model_management:
+    finders:
+        <finder name>:
+            type: MULTI
+            config:
+                # Sequence of other finder names to use in priority order
+                finder_sequence:
+                    - other_finder1
+                    - other_finder2
+"""
+# Standard
+from typing import Optional
+
+# First Party
+import aconfig
+import alog
+
+# Local
+from ...config import get_config
+from ..exceptions import error_handler
+from ..modules import ModuleConfig
+from .model_finder_base import ModelFinderBase
+
+# NOTE: Top-level import done so that global MODEL_MANAGER can be used at
+#   construction time without incurring a circular dependency
+import caikit.core
```

Comment:
Is our goal to eventually kill the global singletons like the `caikit.core.MODEL_MANAGER`? And instead do some sort of dependency injection where it's needed?

Reply:
yes! As I've been working around these things in all of our tests, I'm starting to imagine a future where there is a `CaikitContext` that holds the "globals" as instance variables. We'd then have a special `contextmanater` that would allow a fresh context to replace the default one.

Reply:
or we pick something from here instead of rolling our own? https://wasinski.dev/comparison-of-dependency-injection-libraries-in-python/

Reply:
Yeah, I would like to rent a big cabin in the woods where we can all go crash on this for a couple days and just get it done